qutip/core/operators.py

```diff
@@ -318,10 +328,11 @@ def sigmap():
      [ 0.  0.]]
 
     """
-    return _SIGMAP.copy()
+    dtype = dtype or settings.core["default_dtype"] or _data.CSR
```

Comment:
If used by themselves, dense would be right, but they are often used as building block of larger operator and the dense representation is sticky: binary operation between dense and sparse usually return dense. I though this would make a lot of system dense unless the user does some conversions, which is not what we want.
CSR felt safer in that case.

Reply:
Sounds okay to me. Users can always override the default with `settings.core["default_dtype"]`.